standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/PerfLogger.java

```diff
@@ -186,24 +187,19 @@ public Long getDuration(String method) {
   private transient Timer.Context totalApiCallsTimerContext = null;
 
   private void beginMetrics(String method) {
-    Timer timer = Metrics.getOrCreateTimer(MetricsConstants.API_PREFIX + method);
-    if (timer != null) {
-      timerContexts.put(method, timer.time());
-    }
-    timer = Metrics.getOrCreateTimer(MetricsConstants.TOTAL_API_CALLS);
-    if (timer != null) {
-      totalApiCallsTimerContext = timer.time();
-    }
+    Optional.ofNullable(Metrics.getOrCreateTimer(MetricsConstants.API_PREFIX + method))
+        .ifPresent(timer -> timerContexts.put(method, timer.time()));
+    Optional.ofNullable(Metrics.getOrCreateTimer(MetricsConstants.TOTAL_API_CALLS))
+        .ifPresent(timer -> {totalApiCallsTimerContext = timer.time();});
+    Optional.ofNullable(Metrics.getOrCreateCounter(MetricsConstants.ACTIVE_CALLS + method))
+        .ifPresent(counter -> counter.inc());
```

Comment:
I'm usually not agains changes like this - however this seems like a pretty hot method:

the old implementation have only created 1 new string objects which will be garbage collected - however the new method creates 3 sets of `Nullable` objects and some function objects to make the call...

can we add the `ACTIVE_CALLS` stuff the boring way?